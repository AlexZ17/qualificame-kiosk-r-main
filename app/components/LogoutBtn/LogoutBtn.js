import React, { PureComponent } from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import PropTypes from 'prop-types';

import styles from './LogoutBtnStyle';
import { Feather } from '@expo/vector-icons';

export default class LogoutBtn extends PureComponent {
	render() {
		return (
			<TouchableOpacity
				{...this.props}
			>
				<Feather 
					name={this.props.iconName} 
					size={Platform.OS === 'ios' ? 35:30} style={styles.black}
				/>
			</TouchableOpacity>
		);
	}
}

LogoutBtn.propTypes = {
	iconName: PropTypes.string
}

LogoutBtn.defaultProps = {
	iconName: 'log-out'
}
