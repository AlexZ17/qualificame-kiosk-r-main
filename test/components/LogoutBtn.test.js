import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity, Platform } from 'react-native';

import LogoutBtn from 'qualificame-kiosk/app/components/LogoutBtn';
import { Feather } from '@expo/vector-icons';


function createTestProps (props) {
	return {
		onPress: ()=>{},
		...props,
	}
}

describe('rendering', () => {
	let wrapper = shallow(<LogoutBtn />);
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('displays feather icon', () => {
		expect(wrapper.find(Feather).exists()).toBeTruthy();
	});
	it('the feather icon display an camera', () => {
		expect(wrapper.find(Feather).prop('name')).toEqual('log-out');
	});
});

describe('interactions', () => {
	let wrapper;
	const createWrapper = props => shallow(<LogoutBtn {...props}/>);
	describe('when onPress function is passed to props', () => {
		let fn;
		beforeEach(() => {
			fn = jest.fn();
			const props = createTestProps({ onPress: fn })
			wrapper = createWrapper(props);
		});
		it('it is called on button press', () => {
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(fn).toBeCalled();
		});
	});
})