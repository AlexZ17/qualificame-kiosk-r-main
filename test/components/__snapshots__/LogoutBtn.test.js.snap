// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Icon
    allowFontScaling={false}
    name="log-out"
    size={35}
    style={
      Object {
        "color": "#364350",
      }
    }
  />
</TouchableOpacity>
`;
